dev-libs/libgit2/libgit2-0.26.2.recipe

```diff
@@ -12,19 +12,23 @@ DESCRIPTION="The library provides:
 * thread safety and reentrancy
 * descriptive and detailed error messages
 * ...and more (over 175 different API calls)"
-HOMEPAGE="http://libgit2.github.com/"
-COPYRIGHT="2005-2017 libgit2 project"
+HOMEPAGE="https://libgit2.github.com/"
+COPYRIGHT="2005-2018 libgit2 project"
 LICENSE="GNU LGPL v2"
 REVISION="1"
 SOURCE_URI="https://github.com/libgit2/libgit2/archive/v$portVersion.tar.gz"
-CHECKSUM_SHA256="6a62393e0ceb37d02fe0d5707713f504e7acac9006ef33da1e88960bd78b6eac"
+CHECKSUM_SHA256="747b47d5b02a2387ff81301c694763785181b895690b6eb91ed1ae4b7904307b"
+SOURCE_FILENAME="libgit2-$portVersion.tar.gz"
 
 ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
 
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%.*}"
```

Comment:
The name of the symlink is not `libgit2.so.0` but `libgit2.so.26`:
```
~> ls -go /system/lib/libgit2.so.26 
lr-xr-xr-x 1 0 Mar 11 13:53 /system/lib/libgit2.so.26 -> libgit2.so.0.26.2

~> readelf -d /system/lib/libgit2.so.26 | grep SONAME
 0x000000000000000e (SONAME)             Library soname: [libgit2.so.26]
```
Should I change `libVersionCompat`?
```diff
-libVersionCompat="$libVersion compat >= ${libVersion%.*}"
+libVersionCompat="$libVersion compat >= 26"
```

Remarks:
* libgit2 is used by `kde-frameworks/ktexteditor/` and by 2 of the recipes of `dev-lang/rust/`: `1.21~dev` and `1.22~dev`
* I have already rebuilt libgit2 with these changes:
  * in `BUILD()`: `cmake --build .` instead of `make $jobArgs`
  * in `INSTALL()`: `cmake --build . --target install` instead of `make install`

in the hope that it would create `libgit2.so.26.2` but it still creates libgit2.so.0.26.2`.

Or should we rename the symlink to `libgit2.so.0.26` (and keep `libVersionCompat` unchanged?

Reply:
> Or should we rename the symlink to libgit2.so.0.26 (and keep libVersionCompat unchanged?

Hmm, I think the answer to my own question is: no, because the symlink needs to match the soname.
So I guess what we need to do is:
* `libVersionCompat="$libVersion compat >= 26"`
* bump `REVISION` for the 3 recipes that need libgit2, once we have merged this PR.

Is this correct?

Reply:
Sure, please make sure that the recipes you mentioned were built by our buildbots before bumping the revision.